packages/remix-server-runtime/__tests__/server-test.ts

```diff
@@ -1744,10 +1707,20 @@ describe("shared server runtime", () => {
       expect(indexLoader.mock.calls.length).toBe(0);
 
       let calls = build.entry.module.default.mock.calls;
-      expect(calls.length).toBe(2 * DATA_CALL_MULTIPIER);
-      expect(spy.console.mock.calls.length).toBe(1 * DATA_CALL_MULTIPIER);
+      expect(calls.length).toBe(2);
+      expect(spy.console.mock.calls[0][0].data).toEqual(
+        'Error: No route matches URL "/"'
+      );
+      expect(spy.console.mock.calls[1][0].message).toEqual(
+        "thrown from handleDocumentRequest and expected to be logged in console only once"
+      );
+      expect(spy.console.mock.calls[2][0].message).toEqual(
+        "second error thrown from handleDocumentRequest"
+      );
+      expect(spy.console.mock.calls.length).toBe(3);
```

Comment:
This tests gets updated now that we're properly rendering these from the server request handle and not from the render cycle